### DISCUSSION TITLE
Welcome to SimSharp Discussions!

### DISCUSSION THREAD

**Discussion by abeham** [abeham/SimSharp#36] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nikkjazz** [comment]
Hi
A newbie question here - I'm looking to build a simple drag-and-drop based interface for simsharp that will allow us to build and configure environments, models and data in a web interface.  Before getting started I thought it may be a good ideas to ask if there's any open source examples available we can leverage, alternatively if there's anyone on the forum who has built a similar solution we can collaborate with.

Thanks

**Reply by abeham** [reply]
Interesting approach, I have not heard of something like that for these kinds of simulations. In some way it will probably be similar to other (commercial) tools like AnyLogic, Arena, FlexSim, PlantSim, etc. You'll probably need a code editor. Look into integrating Monaco (e.g. used in VS Code) into your web interface.

**Reply by GR8DAN** [reply]
Like a web version of [Jaam Sim](https://jaamsim.com/). I've been wondering if SVG with its animation options and metadata support could be leveraged for web-based visualisation of event simulators. However, linking parts of the SVG-based images to the event code will be a challenge as Javascript is the Web's language. There are SVG Javascript libraries that exist. Maybe combine something like [Snap.svg](http://snapsvg.io/) with [NoFlo](https://noflojs.org/) and linked back to a C# server application.

**Reply by abeham** [reply]
Regarding visualization, there is Klara's fork of SimSharp where she added the possibility to do visualization and animation with Sim#: https://github.com/KlaraDurst/SimSharp

**Reply by nikkjazz** [reply]
Thanks - I'll look into it.

Cheers
